### PR TITLE
[NO-ISSUE] Fix generate-deploy on Mac OS

### PR DESCRIPTION
**Pull request by brusdev**

### CODE REVIEW

**Comment by brusdev** [comment]
@kevinfrommelt I created this PR to fix the generate-deploy on Mac OS, could you review it?

File: hack/static_manifest_gen.sh [hunk 1]
```diff
@@ -47,8 +47,8 @@ function writeFile() {
         createFile "$destdir/role.yaml"
         true_kind="ClusterRole"
         createFile "$destdir/cluster_role.yaml"
-        sed -i 's/kind: Role/kind: ClusterRole/' \
-          "$destdir/cluster_role.yaml"
+        sed -i.bak 's/kind: Role/kind: ClusterRole/' \
+          "$destdir/cluster_role.yaml" && rm "$destdir/cluster_role.bak"
```
Comment (kevinfrommelt) [inline hack/static_manifest_gen.sh:51]:
```suggestion
          "$destdir/cluster_role.yaml" && rm "$destdir/cluster_role.yaml.bak"
```

File: hack/static_manifest_gen.sh [hunk 2]
```diff
@@ -61,9 +61,9 @@ function writeFile() {
         createFile "$destdir/role_binding.yaml"
         true_kind="ClusterRoleBinding"
         createFile "$destdir/cluster_role_binding.yaml"
-        sed -i -e 's/kind: Role/kind: ClusterRole/' \
+        sed -i.bak -e 's/kind: Role/kind: ClusterRole/' \
           -e 's/kind: RoleBinding/kind: ClusterRoleBinding/' \
-          "$destdir/cluster_role_binding.yaml"
+          "$destdir/cluster_role_binding.yaml" && rm "$destdir/cluster_role_binding.bak"
```
Comment (kevinfrommelt) [inline hack/static_manifest_gen.sh:66]:
```suggestion
          "$destdir/cluster_role_binding.yaml" && rm "$destdir/cluster_role_binding.yaml.bak"
```

**Review by kevinfrommelt — COMMENTED** [review]
The file names need to be updated in the `rm` commands.
```sh
» make bundle generate-deploy
...
» git status
HEAD detached at brusdev/fix_generate_deploy_macos
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	deploy/cluster_role.yaml.bak
	deploy/cluster_role_binding.yaml.bak
```
Otherwise works as expected. I did not have any changes to the cluster role files after running `make bundle generate-deploy`

**Comment by kevinfrommelt** [comment]
Another option is to do `sed -i '' <args>`. In this case there is no backup file created.

**Comment by brusdev** [comment]
@kevinfrommelt thanks I applied your suggested fixes. I didn't use the option `sed -i '' <args>` because it doesn't work with [GNU sed](https://www.gnu.org/software/sed/).

**Pull request merged**